test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -3821,14 +3821,19 @@ def forward(self, input, other):
         y = torch.randn(1, 2, 1)
         self.run_test(FModModel(), (x, y))
 
+    # Note: Integral Tensor fmod to floating-point scalar (not Tensor)
+    # can not be exported, since its type promotion logic is dependent
+    # on knowing the scalar types.
     @skipIfUnsupportedMinOpsetVersion(10)
     def test_fmod_scalar(self):
         class FModModel(torch.nn.Module):
-            def forward(self, input):
-                return torch.fmod(input, 2.55)
+            def forward(self, input, mod):
```

Comment:
This seems will break existing onnx use case. I am wondering whether we still make the original use case working? cc: @spandantiwari, @BowenBao

Reply:
@ejguan could you try removing the check for `Mod` at here https://github.com/pytorch/pytorch/blob/dee82ef3ea1ed2cae4e891f071adff03cb4a6e0d/torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp#L204 and see if it works for the original test case? The was workaround of onnx export for fmod since it was not adhering to type promotion before.

Reply:
Makes sense! Thanks.